packages/toolkit/src/tests/configureStore.test.ts

```diff
@@ -15,45 +16,19 @@ vi.doMock('redux', async () => {
   return redux
 })
 
-vi.doMock('@internal/devtoolsExtension', async () => {
-  const devtools: typeof DevTools = await vi.importActual(
-    '@internal/devtoolsExtension'
-  )
-  vi.spyOn(devtools, 'composeWithDevTools') // @remap-prod-remove-line
-  return devtools
-})
-
-function originalReduxCompose(...funcs: Function[]) {
-  if (funcs.length === 0) {
-    // infer the argument type so it is usable in inference down the line
-    return <T>(arg: T) => arg
-  }
-
-  if (funcs.length === 1) {
-    return funcs[0]
+declare global {
+  interface Window {
+    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__: MockInstance<
+      Parameters<typeof DevTools.composeWithDevTools>,
+      ReturnType<typeof DevTools.composeWithDevTools>
+    >
   }
-
-  return funcs.reduce(
-    (a, b) =>
-      (...args: any) =>
-        a(b(...args))
-  )
-}
-
-function originalComposeWithDevtools() {
-  if (arguments.length === 0) return undefined
-  if (typeof arguments[0] === 'object') return originalReduxCompose
-  return originalReduxCompose.apply(null, arguments as any as Function[])
 }
 
 describe('configureStore', async () => {
-  // RTK's internal `composeWithDevtools` function isn't publicly exported,
-  // so we can't mock it. However, it _does_ try to access the global extension method
-  // attached to `window`. So, if we mock _that_, we'll know if the enhancer ran.
-  const mockDevtoolsCompose = vi
-    .fn()
-    .mockImplementation(originalComposeWithDevtools)
-  ;(window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ = mockDevtoolsCompose
+  const composeWithDevToolsSpy = vi.spyOn(DevTools, 'composeWithDevTools')
+
+  window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ = composeWithDevToolsSpy
```

Comment:
```suggestion
```
since we're now spying on composeWithDevTools directly, we don't need to mock the window at all - we can just let it default to the stub and spy on that instead

Reply:
github won't let me make a suggestion removing the declare global lines but those can obviously go too